tracing-subscriber/src/registry/stack.rs

```diff
@@ -58,3 +64,27 @@ impl SpanStack {
 thread_local! {
     static CONTEXT: RefCell<SpanStack> = RefCell::new(SpanStack::new());
 }
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use super::*;
+
```

Comment:
it would be nice to have a (more integration-y) test someplace with the repro code from #495, and a test layer that asserts spans are closed in the right order. since this issue is relatively simple, i would be fine with merging this PR without it, but it would be nice to have.

Reply:
Completely agree. Will add.